src/server/services/input-types.js

```diff
@@ -441,14 +441,71 @@ defineType({
     },
 });
 
+const TIME_MS = 1;
+const TIME_SEC = 1000 * TIME_MS;
+const TIME_MIN = 60 * TIME_SEC;
+const TIME_HR = 60 * TIME_MIN;
+const TIME_DAY = 24 * TIME_HR;
+const TIME_WEEK = 7 * TIME_DAY;
+const TIME_UNITS_MS = {
+    'ms': TIME_MS, 'msec': TIME_MS, 'msecs': TIME_MS, 'millisecond': TIME_MS, 'milliseconds': TIME_MS,
+    's': TIME_SEC, 'sec': TIME_SEC, 'secs': TIME_SEC, 'second': TIME_SEC, 'seconds': TIME_SEC,
+    'm': TIME_MIN, 'min': TIME_MIN, 'mins': TIME_MIN, 'minute': TIME_MIN, 'minutes': TIME_MIN,
+    'h': TIME_HR, 'hr': TIME_HR, 'hrs': TIME_HR,  'hour': TIME_HR, 'hours': TIME_HR,
+    'd': TIME_DAY, 'day': TIME_DAY, 'days': TIME_DAY,
+    'w': TIME_WEEK, 'week': TIME_WEEK, 'weeks': TIME_WEEK,
+};
+
 defineType({
     name: 'Date',
     description: 'A calendar date.',
-    baseType: 'Any',
+    baseType: 'String',
     parser: input => {
-        input = new Date(input);
-        if (isNaN(input.valueOf())) throw GENERIC_ERROR;
-        return input;
+        const grabNumber = input => {
+            const res = input.match(/^\s*([+-]?)\s*(\d+\.?\d*)\s*$/);
+            return res ? +`${res[1]}${res[2]}` : null;
+        };
+        const basicParser = (input, orElse) => {
+            const num = grabNumber(input);
+            const res = new Date(num !== null ? num : input);
+            if (!isNaN(+res)) return res;
+            if (orElse !== undefined) return orElse;
+            throw GENERIC_ERROR;
+        };
+
+        const MAX_DATE_LEN = 128;
+        let dateLen = Math.min(MAX_DATE_LEN, input.length);
```

Comment:
Not a huge deal but it might be nice to throw a hard error when the string is too long rather than just ignoring the later portions. Perhaps this could be removed if we just subclassed `BoundedString` instead.